frontend/src/components/VisitRouter/index.js

```diff
@@ -38,13 +41,30 @@ const VisitRouter = observer(() => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  const handleSubmitVisit = () => {
+  const handleSubmitVisit = async () => {
     // TODO: logic protecting against concurrent queue entries
     // if existing visit we are coming from QueueTable, so update visit
-    if (existingVisit.id) {
-      participantStore.updateVisit()
-    } else {
-      participantStore.createVisit()
+    try {
+      let isValid = await validateVisitForm(existingVisit)
+      if (existingVisit.id) {
+        !isValid.length
+          ? participantStore.updateVisit()
+          : isValid.map(error =>
+              participantStore.setSnackbarState(
+                `Theres an error in the ${error.name} field.`
+              )
+            )
+      } else {
+        !isValid.length
+          ? participantStore.createVisit()
+          : isValid.map(error =>
+              participantStore.setSnackbarState(
+                `Theres an error in the ${error.name} field.`
+              )
+            )
+      }
+    } catch (err) {
+      handleError(err)
```

Comment:
handleError returns a string. you should set the snackbar state to this returned string

Reply:
Addressed in the latest commit.